FerramAerospaceResearch.Base/UnityJobs/OcclusionJobs.cs

```diff
@@ -0,0 +1,381 @@
+﻿using System;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Jobs;
+using Unity.Mathematics;
+using UnityEngine;
+
+namespace FerramAerospaceResearch.UnityJobs
+{
+    [BurstCompile]
+    public struct SphereDistanceInfo : IComparable<SphereDistanceInfo>, IEquatable<SphereDistanceInfo>
+    {
+        public quaternion q;
+        public float distance;
+
+        public int CompareTo(SphereDistanceInfo obj)
+        {
+            return distance.CompareTo(obj.distance);
+        }
+
+        public bool Equals(SphereDistanceInfo other)
+        {
+            return q.Equals(other.q) && distance.Equals(other.distance);
+        }
+    }
+
+    public struct EMPTY_STRUCT { }
+
+    //http://extremelearning.com.au/how-to-evenly-distribute-points-on-a-sphere-more-effectively-than-the-canonical-fibonacci-lattice/
+    [BurstCompile]
+    public struct SpherePointsJob : IJobParallelFor
+    {
+        [ReadOnly] public int points;
+        [ReadOnly] public float epsilon;
+        [WriteOnly] public NativeArray<quaternion> results;
+        public void Execute(int index)
+        {
+            float goldenRatio = (1 + math.sqrt(5)) / 2;
+            float thetaConstantTerm = 2 * math.PI / goldenRatio;
+            if (index == 0)
+            {
+                results[index] = quaternion.identity;
+            }
+            else if (index == points - 1)
+            {
+                results[index] = quaternion.Euler(180, 0, 0);
+            }
+            else
+            {
+                float theta = index * thetaConstantTerm;
+                float denom = points - 1 + (2 * epsilon);
+                float num = 2 * (index + epsilon);
+                float phi = math.acos(1 - num / denom);
```

Comment:
`denom` can be precomputed in the constuctor as a reciprocal to avoid division

Reply:
Implemented in 53809f6